llama.cpp

```diff
@@ -409,6 +419,11 @@ static bool llama_model_load(
         fin.rdbuf()->pubsetbuf(f_buf.data(), f_buf.size());
         fin.seekg(file_offset);
 
+        // stat the file for file size
+        struct stat st;
+        stat(fname_part.c_str(), &st);
+        const size_t file_size = st.st_size;
+
```

Comment:
Is this portable? Wasn't there some way to determine size using `std::seekg(..)`?